apps/projects/app/components/Shared/FilterBar/FilterBar.js

```diff
@@ -353,27 +351,34 @@ const FilterBar = ({
     )
   }
 
-  const SearchInput = () => (
-    <TextInput
-      placeholder="Search"
-      type="search"
-      onChange={updateTextFilter}
-      value={textFilter}
-      adornment={
-        textFilter === '' && (
-          <IconSearch
-            css={`
-              color: ${theme.surfaceOpened};
-              margin-right: 8px;
-            `}
-          />
-        )
-      }
-      adornmentPosition="end"
-      css="width: 256px"
-      ref={searchRef}
-    />
-  )
+  const SearchInput = () => {
+    layoutName === 'large' ?
+      useEffect(() => { textFilter && searchRef.current && searchRef.current.focus()}) :
+      useEffect(() => { searchRef.current && searchRef.current.focus()})
```

Comment:
The idea was that we only refocus the TextInput in the large layout if a user is already typing in it. Otherwise it will be focused on the initial loading of the page which we don't want. However, in smaller layouts, the TextInput is hidden inside a Popover. When the Popover is loaded, we want the TextInput to be focused right away.

However, I take your point and will try to clean it up.